### ISSUE TITLE
Facing error: ModuleNotFoundError: No module named 'pyodbc', when I run the python based BAT file comprising of pyodbc

### ISSUE THREAD

**Issue by prateekralhan** [mkleehammer/pyodbc#768]
### Environment
- Python: 3.6.1 32bit
- pyodbc: 4.0.30
- OS: Win10 Version 1709
- DB: SQL Server

### Issue
I had installed pyodbc - 4.0.30 in my Win10 OS using pip install pyodbc. It got installed successfully and works fine when I execute the python script. But I want to run my script periodically, so I created an executable batch file (.bat) which can be run by a single click to invoke and run the python script. 

But I am facing this error whenever I execute the batch file. I tried uninstalling and reinstalling. Additionally I also have dependencies installed as well.

![PYODBC_error](https://user-images.githubusercontent.com/29462447/82811546-a7817a00-9eae-11ea-9479-1e8efd1d67f0.png)

**Comment by v-chojas** [comment]
Does the fact that you mention your Python is 3.6.1, but the image shows a path that suggests 3.7, mean that you are not running the correct Python?

**Comment by keitherskine** [comment]
You should be able to run `py -0` to see all the versions of Python installed (globally) on your PC.  Then you can run `py -3.6 -m pip freeze` and `py -3.7 -m pip freeze` to see what modules have been installed into each Python version.

Then you can run the appropriate Python interpreter, e.g. `py -3.6 your_python_code.py`

("py", see [here](https://docs.python.org/3/using/windows.html#from-the-command-line), is a handy utility script installed when you install Python.)

**Comment by prateekralhan** [comment]
@keitherskine and @v-chojas , Thanks a ton for your input !! I completely overlooked this thing that I have python3.6 and python3.7 - both of them installed. The package was installed in one but the BAT file was pointing at the other version.

Thanks again 👍 
Such a silly mistake of mine 👎

**Comment by EmmanuelManana** [comment]
this was very helpful thank you

**Comment by ebouhendi** [comment]
Hi,

I have the same issue. I have only Python 3.9 and I have installed pyodbc 4.0.30. Following is my `pip freeze` output:

click==7.1.2
Flask==1.1.2
itsdangerous==1.1.0
Jinja2==2.11.3
MarkupSafe==1.1.1
pyodbc==4.0.30
Werkzeug==1.0.1

Can you please help me to solve the issue.

**Comment by grdavey** [comment]
For MacOS, I was having this same problem. I upgraded to Python 3.10 and then pyodbc would not work on a fairly large application. I found this thread that explained pyodbc did not work with Python 3.10

macOS
From https://github.com/mkleehammer/pyodbc/issues/995
You'll need the ODBC header files from unixODBC:

brew install unixodbc
export LDFLAGS="-L/opt/homebrew/Cellar/unixodbc/2.3.9_1/lib"
export CPPFLAGS="-I/opt/homebrew/Cellar/unixodbc/2.3.9_1/include"
pip3 install pyodbc

--- THIS DID NOT WORK EITHER---
I finally deleted Python 3.10 from Applications - Initially this did not work, until I went into 
/Library/Frameworks/Python.framework/Versions/3.10.  and deleted the whole directory. 

This may not be a good solution, but, after a day of frustration trying things that didn't work. This did.
